cloud/scope/common.go

```diff
@@ -44,19 +44,13 @@ func CreateLinodeClient(apiKey string, timeout time.Duration) (*linodego.Client,
 	return &linodeClient, nil
 }
 
-func getCredentialDataFromRef(ctx context.Context, crClient K8sClient, credentialsRef corev1.SecretReference, defaultNamespace string) ([]byte, error) {
+func getCredentialDataFromRef(ctx context.Context, crClient K8sClient, credentialsRef corev1.SecretReference, defaultNamespace string) (*corev1.Secret, error) {
```

Comment:
I think we'll want to change the name of this function since we're getting the Secret now instead of specific secret data. Alternatively we could pass which data field to try to extract so this still returns []byte instead of a Secret